src/Microsoft.Restier.WebApi/HttpConfigurationExtensions.cs

```diff
@@ -44,6 +47,11 @@ public static class HttpConfigurationExtensions
         {
             Ensure.NotNull(apiFactory, "apiFactory");
 
+            ApiConfiguration.Configure<TApi>(services =>
+            {
+                services.AddScoped<ODataCountOption>()
```

Comment:
The additional total count request is transferred via a scoped service. It's also could be done by adding subclasses to QueryRequest/QueryResult and then some members, but I'd like to make it really cross-cutting, and so possibly be reused independently. If you don't like it, I'll change to use inheritance.